packages/node/src/indexer/project.service.ts

```diff
@@ -104,80 +108,22 @@ export class ProjectService {
   }
 
   private async ensureProject(): Promise<string> {
-    let schema = await this.getExistingProjectSchema();
+    let schema = await getExistingProjectSchema(
+      this.nodeConfig,
+      this.sequelize,
+      this.subqueryRepo,
+      logger,
+    );
     if (!schema) {
       schema = await this.createProjectSchema();
-    } else {
-      if (argv['force-clean']) {
-        try {
-          // drop existing project schema and metadata table
-          await this.sequelize.dropSchema(`"${schema}"`, {
-            logging: false,
-            benchmark: false,
-          });
-
-          // remove schema from subquery table (might not exist)
-          await this.sequelize.query(
-            ` DELETE
-              FROM public.subqueries
-              WHERE name = :name`,
-            {
-              replacements: { name: this.nodeConfig.subqueryName },
-              type: QueryTypes.DELETE,
-            },
-          );
-
-          logger.info('force cleaned schema and tables');
-
-          if (fs.existsSync(this.nodeConfig.mmrPath)) {
-            await fs.promises.unlink(this.nodeConfig.mmrPath);
-            logger.info('force cleaned file based mmr');
-          }
-        } catch (err) {
-          logger.error(err, 'failed to force clean');
-        }
-        schema = await this.createProjectSchema();
-      }
     }
-
     this.eventEmitter.emit(IndexerEvent.Ready, {
       value: true,
     });
 
     return schema;
   }
 
-  // Get existing project schema, undefined when doesn't exist
-  private async getExistingProjectSchema(): Promise<string> {
```

Comment:
I think we can keep this function but call the implementation in utils, it avoids having to figure out what args to parse where its used